master/internal/rm/kubernetesrm/job.go

```diff
@@ -143,284 +140,363 @@ func newPod(
 		masterTLSConfig:      masterTLSConfig,
 		loggingTLSConfig:     loggingTLSConfig,
 		loggingConfig:        loggingConfig,
-		slots:                msg.Slots,
+		numPods:              msg.NumPods,
+		slotsPerPod:          msg.Slots,
 		podInterface:         podInterface,
 		configMapInterface:   configMapInterface,
 		resourceRequestQueue: resourceRequestQueue,
-		podName:              uniqueName,
+		jobName:              uniqueName,
 		configMapName:        uniqueName,
-		container:            podContainer,
+		podNodeNames:         make(map[string]string),
+		container: cproto.Container{
+			ID:          cproto.ID(msg.Spec.ContainerID),
+			State:       cproto.Assigned,
+			Description: msg.Spec.Description,
+		},
 		containerNames:       containerNames,
 		scheduler:            scheduler,
 		slotType:             slotType,
 		slotResourceRequests: slotResourceRequests,
-		syslog: logrus.New().WithField("component", "pod").WithFields(
+		syslog: logrus.New().WithField("component", "job").WithFields(
 			logger.MergeContexts(msg.LogContext, logger.Context{
-				"pod": uniqueName,
+				"job": uniqueName,
 			}).Fields(),
 		),
 	}
 	return p
 }
 
-func (p *pod) start() error {
-	if p.restore {
-		if p.container.State == cproto.Running {
-			err := p.startPodLogStreamer()
-			if err != nil {
+func (j *job) start() error {
+	if j.restore {
+		if j.container.State == cproto.Running {
+			if err := j.startPodLogStreamers(); err != nil {
```

Comment:
`startJobLogStreamer`?

Reply:
we can't stream job logs (there is no such api) so we start a streamer for each job pod still.